src/submission/validation-clinical/therapy.ts

```diff
@@ -17,42 +17,234 @@
-import { DeepReadonly } from 'deep-freeze';
-import { Donor, Treatment } from '../../clinical/clinical-entities';
-import * as utils from './utils';
-import _ from 'lodash';
 import { getSingleClinicalObjectFromDonor } from '../../common-model/functions';
+import { donorDao } from '../../clinical/donor-repo';
+import { ClinicalInfo, Donor, Treatment } from '../../clinical/clinical-entities';
+import { ClinicalQuery } from '../../clinical/clinical-api';
 import { isValueEqual } from '../../utils';
 
 export const validate = async (
   therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
   existentDonor: DeepReadonly<Donor>,
   mergedDonor: Donor,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
 ): Promise<SubmissionValidationOutput> => {
   // ***Basic pre-check (to prevent execution if missing required variables)***
   if (!therapyRecord || !mergedDonor || !existentDonor) {
     throw new Error("Can't call this function without a registerd donor & therapy record");
   }
 
-  const errors: SubmissionValidationError[] = [];
+  let errors: SubmissionValidationError[] = [];
   const treatment = getTreatment(therapyRecord, mergedDonor, errors);
   if (!treatment) return { errors };
-  checkTreatementHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+  // TODO: Refactor use of push
+  checkTreatmentHasCorrectTypeForTherapy(therapyRecord, treatment, errors);
+
+  const crossFileErrors = await crossFileValidator(
+    mergedDonor,
+    treatment,
+    therapyRecord,
+    submittedRecords,
+  );
+
+  errors = [...errors, ...crossFileErrors];
   return { errors };
 };
 
-export function checkTreatementHasCorrectTypeForTherapy(
+const crossFileValidator = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  // submitted records needed to validate current submission
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const isRadiationRecord = Object.values(RadiationFieldsEnum).some(field =>
+    Object.keys(therapyRecord).includes(field),
+  );
+
+  const radiationErrors: SubmissionValidationError[] = isRadiationRecord
+    ? await validateRadiationRecords(donor, treatment, therapyRecord, submittedRecords)
+    : [];
+
+  const crossFileErrors = [...radiationErrors];
+
+  return crossFileErrors;
+};
+
+interface DonorResult {
+  donors: DeepReadonly<Donor>[];
+}
+
+const emptyDonorResult: DonorResult = { donors: [] };
+
+const validateRadiationRecords = async (
+  donor: Donor,
+  treatment: DeepReadonly<Treatment>,
+  therapyRecord: DeepReadonly<SubmittedClinicalRecord>,
+  submittedRecords: DeepReadonly<ClinicalSubmissionRecordsByDonorIdMap>,
+) => {
+  const { programId } = donor;
+
+  const {
+    clinicalInfo: {
+      submitter_donor_id: relatedTreatmentDonorId,
+      submitter_treatment_id,
+      treatment_type,
+    },
+  } = treatment;
+
+  const {
+    submitter_donor_id: submittedTherapyDonorId,
```

Comment:
is this renaming to provide more context in the file? ie. in the whole system it's `submitter_donor_id` but we are using it here as `submittedTherapyDonorId`?

Reply:
Yes. The intent here is to disambiguate between the Submitter Donor ID associated with the Therapy Record we are currently validating, and any of the multiple Submitter Donor IDs we are matching it against

We are checking, does Submitter Donor ID A match Submitter Donor ID B, C, D, etc., so using the key name is not important necessarily, the end result is just a boolean

Reply:
There might be potential to just lump all comparison values together, i.e. we have our `submitter_donor_id` which is from the submitted therapy record, then we loop through a new `relatedTreatments[]` and compare if `submitter_donor_id === relatedTreatments[i].submitter_donor_id`, does that make sense?

Reply:
Keeping as is for now, can revisit a map solution at a later time